### ISSUE TITLE
Cannot handle new 3.12 f-strings, hangs up completely

### ISSUE THREAD

**Issue by crankedguy** [EeyoreLee/vscode-extension-yapf#66]
Hi,

your extension cannot handle new f-string capabilities of Python 3.12, writing a single line of code like this
`f"https://ldomain.com.com{user["x1"]["x2"]["x3"]["x3"]["x5"]}",`

makes the whole extension break down silently and stops it from any further formatting.
Only if you change the inside " to ' it works again.
But this is perfectly valid now in 3.12 and a massive improvement for using f-string. Please make it possible that your extension can handle that.. And on top does not just fail silently

Thanks

**Comment by EeyoreLee** [comment]
@crankedguy - yep, this should be added on the google/yapf side. It's issued here [google/yapf](https://github.com/google/yapf/issues/1136). when yapf support this feature, I'll update the bundled yapf.

**Comment by crankedguy** [comment]
thanks for the info. great, only open since two months...

**Comment by Spitfire1900** [comment]
@EeyoreLee The extension should still have error recovery without hanging + report an error to the user w/ output in the Extension's output pane.

**Comment by crankedguy** [comment]
> @EeyoreLee The extension should still have error recovery without hanging + report an error to the user w/ stack trace in the Extension's output pane.

this is one of the main problems that this does not happen

**Comment by EeyoreLee** [comment]
The extension didn't hang up. It's still running, and just do nothing modification if yapf package raise a exception for a file. But it will indeed report an error in yapf extension output channel cause I think most of people won‘t look at it unless the extension doesn't work.

**Comment by crankedguy** [comment]
@EeyoreLee You are right, I oversaw that in the specific output channel, however it is a bit of a burden that the whole file is affected by this,,, you cannot even disable that with # yapf: disable, yapf is just completely unusable for Python 3.12 at all then as far as I can see, if there is no other solution to circumvent that at least for portions of code
